Wootric/WootricSDK-iOS
# Pluralsight master

Creating a new PR with changes proposed in https://github.com/Wootric/WootricSDK-iOS/pull/46 and adding some fixes myself.

Sorry for the big PR, but with the exception of the Logger functionality, it's just updating project settings, removing warnings, etc.

## Changes
- Add WTRLogger to allow setting log level so logs can be disabled
- Add new setters to `Wootric.h` to set the WTRLogger mode
- Update license comment in all files
- Update project settings [1](WootricSDK-Demo/WootricSDK-Demo.xcodeproj/project.pbxproj), [2](WootricSDK/WootricSDK.xcodeproj/project.pbxproj), [3](WootricSDK/WootricSDK.xcodeproj/xcshareddata/xcschemes/WootricSDK.xcscheme)
- Remove `client_secret` var from [ViewController.m](WootricSDK-Demo/WootricSDK-Demo/ViewController.m)
- Remove repeated header in `WootricSDK.podspec`

## How to test
- Checkout this branch
- Use the new setters and check that the logs show the correct info depending on the mode
```objc_c
[Wootric setLogLevelNone];
[Wootric setLogLevelError];
[Wootric setLogLevelVerbose];
```

Test on both smartphone and table devices.

## Trello
There's no Trello card for this. This functionality was kindly added by Pluralsight.

## PR
This PR was created in place of https://github.com/Wootric/WootricSDK-iOS/pull/46

*Note: once the PR is approved and ready to merge I will squash all commits into one and modify the SDK version*
